### ISSUE TITLE
Immediates should be unsigned

### ISSUE THREAD

**Issue by daniel-ziegler** [llvm-dcpu16/llvm-dcpu16#101]
Immediates in the emitted assembly range from -32768 to 32767, rather than 0 to 65535. This breaks all the emulators I've tried, including http://0x10co.de/ .

Example code:

```
    int main(void) {
        int* ptr = (int*)0x8000;
        *ptr = 0x8000;
        return 0;
    }


```

Actual output:

```
    :autoinit       ;;Init data stack register C
        SET I, SP
        SUB I, 256
    :autostart
        JSR main
    :autohalt SET PC, autohalt
        ; .file "fib.c"
        .text
        .globl  main
        ; .align    2
    :main
        SUB I, 4 ; The Notch order
        SET [2+I], 0 ; The Notch order
        SET A, 0 ; The Notch order
        SET [I], -32768 ; The Notch order
        SET [-32768], -32768 ; The Notch order
        ADD I, 4 ; The Notch order
        SET PC, POP ; The Notch order


```

Expected output:

```
    :autoinit       ;;Init data stack register C
        SET I, SP
        SUB I, 256
    :autostart
        JSR main
    :autohalt SET PC, autohalt
        ; .file "fib.c"
        .text
        .globl  main
        ; .align    2
    :main
        SUB I, 4 ; The Notch order
        SET [2+I], 0 ; The Notch order
        SET A, 0 ; The Notch order
        SET [I], 32768 ; The Notch order
        SET [32768], 32768 ; The Notch order
        ADD I, 4 ; The Notch order
        SET PC, POP ; The Notch order


```

**Comment by daniel-ziegler** [comment]
Uh, sorry about the formatting. I can't figure out how to do it right :P.

**Comment by krasin** [comment]
My first reaction was to suggest using unsigned int, but currently I see that it is also broken. Looking into this...

**Comment by ghost** [comment]
We told clang in the target description that we are using signed shorts for pointer. Why don't we just let the ASM Printer print out the Imm as Hex values? (Note to me: the imm value is saved in an int64_t)

**Comment by krasin** [comment]
Fixed by @arbaal in d245c648af6b3d0e562b54a73fc2e86d0cda438b

**Comment by krasin** [comment]
Thx!
